### PR TITLE
[KYUUBI #1988] Tune initialization of vars depend on system level confs

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

The `new SparkConf()` and `KyuubiConf()` executes early than UT, and can not re-initialize after changing system properties, which makes conf overwriting does not work.

After the change, the `sparkConf.setIfMissing("spark.sql.catalogImplementation", defaultCat)` work as expected when we overwriting `spark.sql.catalogImplementation` in UT, and Hudi uses `in-memory` catalog to create table.

Xnip2022-03-02_19-10-06

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
cc @wForget @yaooqinn

**Comment by ulysses-you** [comment]
seems the failed test is relevant

**Review by SteNicholas — APPROVED** [review]
LGTM.

**Comment by ulysses-you** [comment]
does it works ? it seems to me the problem is order of extra config and initialization config. We should setup conf first then overwrite using `withKyuubiConf`. 

See the code in `WithSparkSQLEngine#startSparkEngine`
```scala
......
withKyuubiConf.foreach { case (k, v) =>
  System.setProperty(k, v)
  kyuubiConf.set(k, v)
}

SparkSession.clearActiveSession()
SparkSession.clearDefaultSession()
SparkSQLEngine.setupConf()
spark = SparkSQLEngine.createSpark()
SparkSQLEngine.startEngine(spark)
......
```

**Comment by pan3793** [comment]
I don't think the order is a problem, we use setIfMissing in `SparkSQLEngine#setupConf`

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala [hunk 1]
```diff
@@ -71,41 +71,48 @@ case class SparkSQLEngine(spark: SparkSession) extends Serverable("SparkSQLEngin
...
 
   var currentEngine: Option[SparkSQLEngine] = None
 
-  private val user = currentUser
+  private lazy val user = currentUser
 
   private val countDownLatch = new CountDownLatch(1)
 
+  SignalRegister.registerLogger(logger)
+  setupConf()
```
Comment (wForget) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala:87]: Should the `SparkSQLEngine.setupConf()` call in `WithSparkSQLEngine#startSparkEngine` be removed?

Comment (pan3793) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala:87]: No, after changing system properties, we need re-init the SparkConf and KyuubiConf to let them pick up the system properties

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thank, merging to master

**Pull request closed without merging**